### PR TITLE
feat(QueryEditor): enable explain query modes

### PR DESCRIPTION
**Pull request by artemmufazalov**
Connected PRs:
- #360
- #362
- #363 
- #364 
- #367 
- #369

### CODE REVIEW

File: src/containers/Tenant/QueryEditor/QueryExplain/QueryExplain.js [hunk 1]
```diff
@@ -102,12 +102,6 @@ function QueryExplain(props) {
         };
     }, []);
 
-    useEffect(() => {
```
Comment (artemmufazalov) [inline src/containers/Tenant/QueryEditor/QueryExplain/QueryExplain.js:105]: It occurred, that `explain-ast` is run in data-query mode, so when we request `ast` for `explain-scan` or `explain-query`, we have `ast` that does not correspond to initial mode. Thus, it should be displayed only if query response have it

File: src/containers/Tenant/QueryEditor/QueryEditorControls/QueryEditorControls.scss [hunk 2]
```diff
@@ -0,0 +1,57 @@
+.ydb-query-editor-controls {
```
Comment (artemmufazalov) [inline src/containers/Tenant/QueryEditor/QueryEditorControls/QueryEditorControls.scss:1]: These styles was moved from `QueryEditor.scss` with minor changess (e.g. `&__mode-selector` was added)

File: src/containers/Tenant/QueryEditor/QueryEditor.js [hunk 3]
```diff
@@ -4,25 +4,21 @@ import {connect} from 'react-redux';
...
-import SaveQuery from './SaveQuery/SaveQuery';
 import SavedQueries from './SavedQueries/SavedQueries';
-import {Icon} from '../../../components/Icon';
 import QueryResult from './QueryResult/QueryResult';
 import QueryExplain from './QueryExplain/QueryExplain';
+import {QueryEditorControls} from './QueryEditorControls/QueryEditorControls';
 
 import {
-    sendQuery,
+    sendExecuteQuery,
```
Comment (artemmufazalov) [inline src/containers/Tenant/QueryEditor/QueryEditor.js:17]: It was just renamed

File: src/store/reducers/executeQuery.ts [hunk 4]
```diff
@@ -55,7 +46,6 @@ const initialState = {
                 ? MAXIMUM_QUERIES_IN_HISTORY - 1
                 : queriesHistoryInitial.length - 1,
     },
-    runAction: readSavedSettingsValue(QUERY_INITIAL_RUN_ACTION_KEY, RUN_ACTIONS_VALUES.script),
```
Comment (artemmufazalov) [inline src/store/reducers/executeQuery.ts:58]: Selected query mode is now managed inside `QueryEditor` component

**Comment by Raubzeug** [comment]
When the new mode is enabled, it is unclear, what action is produced by the button `Type: Script | Scan` (actually, no action). I would better make this button similar with query type selection (in the old mode).

0EEB6DB9-FB6B-4BDE-B3E1-871F644EE16D

**Review by Raubzeug — CHANGES_REQUESTED** [review]

File: src/containers/Tenant/QueryEditor/QueryEditor.js [hunk 5]
```diff
@@ -633,22 +593,25 @@ const mapStateToProps = (state) => {
         executeQuery: state.executeQuery,
         explainQuery: state.explainQuery,
         savedQueries: parseJson(getSettingValue(state, SAVED_QUERIES_KEY)),
```
[the remark this answers is not in the document]
Comment (artemmufazalov) [reply src/containers/Tenant/QueryEditor/QueryEditor.js:595]: We don't need to parse value if it used as a string value, so `getSettingValue` doesn't always go with `parseJson` . As for me, adding some additional `getParsedSettingValue` helper will not make code easier as it won't be clear when it should be used and when we should use just `getSettingValue`

[the remark this answers is not in the document]
Comment (Raubzeug) [reply src/containers/Tenant/QueryEditor/QueryEditor.js:595]: I suggested having two helpers! But it's totally up to you!
Unfortunately now it is not clear too - when we should parse JSON, and when value is a plain string... What if add `try ... catch`? If `JSON.parse` fails, return unparsed value, and if succeed - return parsed.

[the remark this answers is not in the document]
Comment (artemmufazalov) [reply src/containers/Tenant/QueryEditor/QueryEditor.js]: Agree with you, now it's unclear too. Added additional parser, maybe I'll replace old one with it when I'll be refactoring

**Review by Raubzeug — APPROVED** [review]

**Review by Raubzeug — APPROVED** [review]

**Pull request merged**